FWCore/Utilities/src/ObjectWithDict.cc

```diff
@@ -74,9 +76,13 @@ namespace edm {
     }
 
     if (to.hasBase(from)) { // down cast
+#ifndef _LIBCPP_VERSION
       // use the internal dynamic casting of the compiler (e.g. libstdc++.so)
       void* address = abi::__dynamic_cast(address_, static_cast<abi::__class_type_info const*>(&from.typeInfo()), static_cast<abi::__class_type_info const*>(&to.typeInfo()), -1);
```

Comment:
MacOS clang doesn't link libc++abi by default. I can try linking it in and see if the original will compile.